client/dev-error-overlay.js

```diff
@@ -0,0 +1,39 @@
+// @flow
+import React from 'react'
+import {applySourcemaps} from './source-map-support'
+import ErrorDebug, {styles} from '../lib/error-debug'
+import type {RuntimeError, ErrorReporterProps} from './error-boundary'
+
+type State = {
+  mappedError: null | RuntimeError
+}
+
+// This component is only used in development, sourcemaps are applied on the fly because componentDidCatch is not async
+class DevErrorOverlay extends React.Component<ErrorReporterProps, State> {
+  state = {
+    mappedError: null
+  }
+
+  componentDidMount () {
+    const {error} = this.props
+
+    // Since componentDidMount doesn't handle errors we use then/catch here
+    applySourcemaps(error).then(() => {
+      this.setState({mappedError: error})
+    }).catch(console.error)
```

Comment:
Maybe here we can set `this.state.mappedError` to the error from applySourcemaps, or atleast set the error to something like "Couldn't render error. Check console for additional debugging details".